packages/nice-grpc-prometheus/src/common.ts

```diff
@@ -11,6 +11,13 @@ export const codeLabel = 'grpc_code';
  * 1ms, 4ms, 16ms, ..., ~1 hour in seconds
  */
 export const latencySecondsBuckets = exponentialBuckets(0.001, 4, 12);
+export const labelNames = [
+  typeLabel,
+  serviceLabel,
+  methodLabel,
+  pathLabel,
+  codeLabel,
+];
```

Comment:
The set of labels is not the same for every metric. Particularly the `code` label is missing in some of them.

Reply:
I thought about it.
1. There is indeed an extra label, but it looks like it has absolutely no effect on behavior.
2. Attempting to separate labels leads to more complexity for the user and more possibility of mistakes.

An alternative way is to create custom wrappers for each metric that encapsulate the necessary parts. But this is too cumbersome.

What are your thoughts on this? Maybe there are some suggestions?

Reply:
Indeed it seems that putting a label to that list does not make it required and does not affect the metrics where we don't use it.

However according to the [code of prom-client](https://github.com/siimon/prom-client/blob/611e228595be2add302d43aebefc762e580d26ce/lib/util.js#L46) it would break if we use the `metric.label(...)` method: it checks that the array lengths are the same.

That said, since we don't use that method, I'm ok with having it this way.

Reply:
I've decided to separate `labelNames` and `labelNamesWithCode` for better reliability. Doesn't look as bad as I expected.